In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/mixalldata_clean.csv")

df.head()

,type,sendTime,sender,senderPseudo,messageID,class,posx,posy,posz,posx_n,...,aclz,aclx_n,acly_n,aclz_n,hedx,hedy,hedz,hedx_n,hedy_n,hedz_n
0,4,72002.302942,130137,101301377,422013806,0,266.982401,32.336955,0.0,3.480882,...,0.0,0.000862,0.000862,0.0,-0.102790,0.994703,0.0,20.038218,17.541001,0.0
1,4,72003.302942,130137,101301377,422023410,0,266.827208,34.624145,0.0,3.546261,...,0.0,0.000107,0.001040,0.0,-0.099856,0.995002,0.0,20.441139,14.467283,0.0
2,4,72004.302942,130137,101301377,422032081,0,266.420297,38.836461,0.0,3.544045,...,0.0,0.000172,0.001661,0.0,-0.099856,0.995002,0.0,20.850473,11.941528,0.0
3,4,72005.302942,130137,101301377,422040712,0,268.912026,45.414229,0.0,3.340080,...,0.0,0.000171,0.001654,0.0,-0.100172,0.994970,0.0,21.323229,9.633965,0.0
4,4,72006.302942,130137,101301377,422052949,0,268.242276,53.729986,0.0,3.328872,...,0.0,0.000193,0.001852,0.0,-0.097105,0.995274,0.0,21.788034,7.824555,0.0


In [2]:
df.shape

(3194808, 30)

In [3]:
df.info

<bound method DataFrame.info of          type      sendTime  sender  senderPseudo  messageID  class  \
0           4  72002.302942  130137     101301377  422013806      0   
1           4  72003.302942  130137     101301377  422023410      0   
2           4  72004.302942  130137     101301377  422032081      0   
3           4  72005.302942  130137     101301377  422040712      0   
4           4  72006.302942  130137     101301377  422052949      0   
...       ...           ...     ...           ...        ...    ...   
3194803     4  79376.752610  140895    1706441661  447808029     19   
3194804     4  79377.252610  140895    1806441661  447808677     19   
3194805     4  79377.752610  140895    1906441661  447809215     19   
3194806     4  79378.252610  140895    2006441661  447809698     19   
3194807     4  79378.752610  140895    2106441661  447810281     19   

                posx        posy  posz    posx_n  ...  aclz    aclx_n  \
0         266.982401   32.336955   0.0  3.

In [4]:
df.columns

Index(['type', 'sendTime', 'sender', 'senderPseudo', 'messageID', 'class',
       'posx', 'posy', 'posz', 'posx_n', 'posy_n', 'posz_n', 'spdx', 'spdy',
       'spdz', 'spdx_n', 'spdy_n', 'spdz_n', 'aclx', 'acly', 'aclz', 'aclx_n',
       'acly_n', 'aclz_n', 'hedx', 'hedy', 'hedz', 'hedx_n', 'hedy_n',
       'hedz_n'],
      dtype='object')

In [5]:
df.describe()

,type,sendTime,sender,senderPseudo,messageID,class,posx,posy,posz,posx_n,...,aclz,aclx_n,acly_n,aclz_n,hedx,hedy,hedz,hedx_n,hedy_n,hedz_n
count,3194808.0,3.194808e+06,3.194808e+06,3.194808e+06,3.194808e+06,3.194808e+06,3.194808e+06,3.194808e+06,3194808.0,3.194808e+06,...,3194808.0,3.194808e+06,3.194808e+06,3194808.0,3.194808e+06,3.194808e+06,3194808.0,3.194808e+06,3.194808e+06,3194808.0
mean,4.0,5.076333e+04,7.383339e+04,9.789703e+07,2.269706e+08,4.842623e+00,5.990197e+02,6.673366e+02,0.0,3.982328e+00,...,0.0,1.001828e-03,1.123134e-03,0.0,-4.058593e-02,3.554393e-02,0.0,1.338068e+01,1.307596e+01,0.0
std,0.0,1.825566e+04,4.204429e+04,3.425484e+08,1.373305e+08,6.731989e+00,3.915811e+02,3.056494e+02,0.0,6.845493e-01,...,0.0,2.853273e-03,2.795307e-03,0.0,6.784653e-01,7.094474e-01,0.0,1.426925e+01,1.422393e+01,0.0
min,4.0,2.406028e+02,9.000000e+00,1.000000e+00,4.184000e+03,0.000000e+00,-2.001299e+01,-2.247956e+01,0.0,0.000000e+00,...,0.0,0.000000e+00,0.000000e+00,0.0,-1.000000e+00,-1.000000e+00,0.0,0.000000e+00,0.000000e+00,0.0
25%,4.0,3.329417e+04,3.794700e+04,1.040041e+07,1.185950e+08,0.000000e+00,2.287022e+02,4.211277e+02,0.0,3.496324e+00,...,0.0,1.201690e-05,1.176814e-05,0.0,-7.049776e-01,-6.618322e-01,0.0,3.696413e+00,3.583056e+00,0.0
50%,4.0,5.050967e+04,7.297500e+04,1.079390e+07,2.237876e+08,0.000000e+00,5.508116e+02,7.045325e+02,0.0,4.003286e+00,...,0.0,1.116123e-04,1.216984e-04,0.0,-1.876019e-02,6.520315e-02,0.0,8.752028e+00,8.212182e+00,0.0
75%,4.0,6.674273e+04,1.103550e+05,1.011660e+08,3.449650e+08,1.200000e+01,9.147023e+02,8.983254e+02,0.0,4.505197e+00,...,0.0,7.687946e-04,1.103780e-03,0.0,6.009444e-01,7.274598e-01,0.0,1.740006e+01,1.711267e+01,0.0
max,4.0,8.639998e+04,1.479810e+05,4.206512e+09,4.586228e+08,1.900000e+01,1.518956e+03,1.522609e+03,0.0,8.245800e+00,...,0.0,1.465100e-01,1.335837e-01,0.0,1.000000e+00,1.000000e+00,0.0,1.213064e+02,1.136965e+02,0.0


## Description du dataset VeReMi Extension

Le dataset contient **3 194 808 observations** et **30 colonnes** décrivant les messages BSM (Basic Safety Messages) échangés entre véhicules dans un réseau VANET simulé.

Les variables incluent :
- `sendTime` : le temps d'envoi du message
- `senderPseudo` : l'identifiant pseudonyme du véhicule émetteur
- `messageID` : l'identifiant unique du message
- La position des véhicules (`posx`, `posy`) et leurs valeurs de bruit (`posx_n`, `posy_n`)
- La vitesse (`spdx`, `spdy`) et leurs valeurs de bruit
- L'accélération (`aclx`, `acly`) et leurs valeurs de bruit
- La direction (`hedx`, `hedy`) et leurs valeurs de bruit

La variable cible est :
- `class` : indique le comportement du véhicule — 0 pour un comportement normal, et 1 à 19 pour les différents types d'attaques

In [6]:
df['class'].value_counts()

class
0     1900539
16     175391
13     131305
15     129270
14     126724
18      86883
19      82100
6       44359
11      44337
17      44310
3       43857
1       43653
2       43567
10      43264
12      43118
9       42790
8       42583
4       42575
7       42258
5       41925
Name: count, dtype: int64

In [7]:
df.isnull().sum()

type            0
sendTime        0
sender          0
senderPseudo    0
messageID       0
class           0
posx            0
posy            0
posz            0
posx_n          0
posy_n          0
posz_n          0
spdx            0
spdy            0
spdz            0
spdx_n          0
spdy_n          0
spdz_n          0
aclx            0
acly            0
aclz            0
aclx_n          0
acly_n          0
aclz_n          0
hedx            0
hedy            0
hedz            0
hedx_n          0
hedy_n          0
hedz_n          0
dtype: int64

In [8]:
df.duplicated().sum()

0

## Analyse des attaques dans le dataset

La variable `class` permet d'identifier le type de comportement de chaque véhicule. 
Le dataset contient 20 classes différentes — une classe normale (class=0) et 19 types 
d'attaques (class=1 à 19).

On observe environ 1 900 539 observations normales contre 1 294 269 observations 
correspondant à des attaques.

Le dataset contient 19 types d'attaques différents avec des nombres d'observations 
variables selon le type.

L'analyse du dataset montre qu'il n'y a pas de valeurs manquantes ni de doublons 
dans les données.